src/packet-ffxiv.c

```diff
@@ -145,8 +150,15 @@ static int dissect_frame(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, vo
     next_tvb = tvb_new_subset_remaining(tvb, FRAME_HEADER_LEN);
   }
 
-  tcp_dissect_pdus(next_tvb, pinfo, frame_tree, TRUE, BLOCK_HEADER_LEN,
-    get_message_length, dissect_message, data);
+  offset = 0;
+  for(int i = 0; i < header.blocks; ++i) {
```

Comment:
Relying on optimisations between 3 separate compilers can result in very unexpected behaviour. I don't mind a for loop but since it isn't used internally for iterating anything, while just seems more appropriate.